File: odlm_app/lib/dto/ReadSeatResponseDto.dart
class ReadSeatResponseDto {
  final int? userId; // null이 될 수 있으므로 int?로 변경
  final bool isUsed;
  final int useCount;
  final int? leaveId; // null이 될 수 있으므로 int?로 변경
  final int duration;
  final int leaveCount;
  final int? maxLeaveCount; // null이 될 수 있으므로 int?로 변경
  final int studyTime;

  ReadSeatResponseDto({
    required this.userId,
    required this.isUsed,
    required this.useCount,
    required this.leaveId,
    required this.duration,
    required this.leaveCount,
    required this.maxLeaveCount,
    required this.studyTime,
  });

  factory ReadSeatResponseDto.fromJson(Map<String, dynamic> json) {
    return ReadSeatResponseDto(
      userId: json['userId'] as int?, // null 허용
      isUsed: json['isUsed'] as bool? ?? false, // 기본값 false
      useCount: json['useCount'] as int? ?? 0, // 기본값 0
      leaveId: json['leaveId'] as int?, // null 허용
      duration: json['duration'] as int? ?? 0, // 기본값 0
      leaveCount: json['leaveCount'] as int? ?? 0, // 기본값 0
      maxLeaveCount: json['maxLeaveCount'] as int?, // null 허용
      studyTime: json['studyTime'] as int? ?? 0, // 기본값 0
    );
  }
}

File: odlm-server/src/main/java/com/project/odlmserver/controller/dto/seat/ReadSeatResponseDto.java
package com.project.odlmserver.controller.dto.seat;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ReadSeatResponseDto {
	private Long userId;

	private Boolean isUsed; // 실제 사용 여부
	private Long useCount = 0L; // 사용 시간 카운트 (분 단위 측정)

	private Long leaveId; //자리비움을 신청한 아이디

	private Long duration = 0L; // 자리 사용 기간 (분 단위)
	private Long leaveCount = 0L; //자리 비움 시간 카운트 (분 단위 측정)
	private Long maxLeaveCount;

	private Long studyTime = 0L;
}
